msal/src/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -345,13 +365,25 @@ private MsalAccessTokenCacheItem FindAccessTokenCommon
                 }
 
                 requestParams.RequestContext.Logger.Info("Matching entry count -" + tokenCacheItems.Count);
-
-                IEnumerable<MsalAccessTokenCacheItem> filteredItems =
-                    tokenCacheItems.Where(
-                            item =>
-                                item.ScopeSet.ScopeContains(requestParams.Scope));
-
-                requestParams.RequestContext.Logger.Info("Matching entry count after filtering by scopes - " + filteredItems.Count());
+    
+
+                IEnumerable<MsalAccessTokenCacheItem> filteredItems =
+                   tokenCacheItems.Where(
+                           item => item.ScopeSet.ScopeContains(requestParams.Scope));
+
+                //Adfs does not return scopes in resource/scope format
+                if (requestParams.Authority.AuthorityType == Core.Instance.AuthorityType.Adfs && filteredItems.Count()==0)
+                {
+                    SortedSet<string> scopes = new SortedSet<string>();
+                    foreach (string scope in requestParams.Scope)
+                    {
+                        scopes.Add(scope.Substring(scope.LastIndexOf("/") + 1));

```

Comment:
>scope.Substring(scope.LastIndexOf("/") + 1) [](start = 35, length = 43)

this looks magical. Can you pls. show an example of what the resource/scope then looks like in ADFS?